interceptor.go

```diff
@@ -78,245 +74,108 @@ func (i *Interceptor) getAndInitInstrument(isClient bool) *instruments {
 
 // WrapUnary implements otel tracing and metrics for unary handlers.
 func (i *Interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
-	return func(ctx context.Context, request connect.AnyRequest) (connect.AnyResponse, error) {
-		requestStartTime := i.config.now()
-		req := &Request{
-			Spec:   request.Spec(),
-			Peer:   request.Peer(),
-			Header: request.Header(),
+	return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		startAt := i.config.now() // TODO: get from connect
+		request := &Request{
+			Spec:   req.Spec(),
+			Peer:   req.Peer(),
+			Header: req.Header(),
 		}
 		if i.config.filter != nil {
-			if !i.config.filter(ctx, req) {
-				return next(ctx, request)
+			if !i.config.filter(ctx, request) {
+				return next(ctx, req)
```

Comment:
Oof, these are kinda terrible names. Simply swapping `request` and `req` doesn't clarify anything IMO. Can we use more descriptive names to distinguish what these two things are?

Reply:
Was trying to keep consistent of what `req` and `request` refer to but yep, will fix.

Reply:
Switched to `call` var but refrained from renaming the type. It is request metadata derived from `connect.Request` objects. Call is better but maybe could be just be called info or metadata?

Reply:
> derived from `connect.Request` objects.

Except for client streams, which can have multiple requests 😛  (hence the name suggestion).

FWIW, I'm fine leaving the type name alone, since it's exported. But in that case, maybe `req` and `msg` would be more clear?

Reply:
I think this is resolve, but please flag if not. Use `call` for `*Request` and `req` `rsp` for connect's Request and Response.